src/libstd/io/impls.rs

```diff
@@ -147,6 +147,11 @@ impl<B: BufRead + ?Sized> BufRead for Box<B> {
 // =============================================================================
 // In-memory buffer implementations
 
+/// Read is implemented for `&[u8]` by copying from the slice.
+///
+/// Note that each read from the slice updates the slice itself so that after
+/// each read, the slice points only to the yet unread part. The slice will be
+/// empty when end of file is reached.
```

Comment:
The `Read` trait calls this situation "end of file" (with quotes) and/or EOF. So I want to keep that terminology. Using either EOF or "end of file" should be good?